tesseract_motion_planners/include/tesseract_motion_planners/simple/profile/simple_planner_profile.h

```diff
@@ -42,26 +42,10 @@ TESSERACT_COMMON_IGNORE_WARNINGS_POP
 
 namespace tesseract_planning
 {
-using JointJointStepGenerator = std::function<CompositeInstruction(const JointWaypoint&,
-                                                                   const JointWaypoint&,
-                                                                   const PlanInstruction&,
-                                                                   const PlannerRequest&,
-                                                                   const ManipulatorInfo&)>;
-using JointCartStepGenerator = std::function<CompositeInstruction(const JointWaypoint&,
-                                                                  const CartesianWaypoint&,
-                                                                  const PlanInstruction&,
-                                                                  const PlannerRequest&,
-                                                                  const ManipulatorInfo&)>;
-using CartJointStepGenerator = std::function<CompositeInstruction(const CartesianWaypoint&,
-                                                                  const JointWaypoint&,
-                                                                  const PlanInstruction&,
-                                                                  const PlannerRequest&,
-                                                                  const ManipulatorInfo&)>;
-using CartCartStepGenerator = std::function<CompositeInstruction(const CartesianWaypoint&,
-                                                                 const CartesianWaypoint&,
-                                                                 const PlanInstruction&,
-                                                                 const PlannerRequest&,
-                                                                 const ManipulatorInfo&)>;
+using SimplePlannerStepGeneratorFn = std::function<CompositeInstruction(const PlanInstruction&,
+                                                                        const PlanInstruction&,
+                                                                        const PlannerRequest&,
+                                                                        const ManipulatorInfo&)>;
```

Comment:
From what I understand, some other class needs to invoke this method to produce a `CompositeInstruction` that is the interpolated path. If that's correct, would it make more sense to create a base class with this method and give a pointer to that base class to the objects that need to use it to interpolate instructions? It seems like this PR adds several helper structs and functions (e.g. `LVSTransitionInfo`) that could be eliminated or better compartmentalized if you created implementations that do specific types of interpolation and contain their own implementation-specific parameters (e.g. longest valid segment length, min points)

Reply:
@marip8 I moved everything from the step generator to the profile classes which simplified things. Let me know what you think.